relay-ourlogs/src/otel_to_sentry.rs

```diff
@@ -395,4 +407,182 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn parse_otel_log_without_span_and_trace_ids() {
+        // Test with completely missing traceId and spanId fields
+        let json = r#"{
+            "timeUnixNano": "1544712660300000000",
+            "observedTimeUnixNano": "1544712660300000000",
+            "severityNumber": 10,
+            "severityText": "Information",
+            "body": {
+                "stringValue": "Log without trace context"
+            },
+            "attributes": [
+                {
+                    "key": "test.attribute",
+                    "value": {
+                        "stringValue": "test value"
+                    }
+                }
+            ]
+        }"#;
+
+        let otel_log: OtelLogRecord = serde_json::from_str(json).unwrap();
+        let our_log: Annotated<OurLog> = Annotated::new(otel_to_sentry_log(otel_log, None, None));
+
+        // Note: trace_id will be a random UUID, so we only check that it exists and has the right metadata
+        let trace_id_value = our_log.value().unwrap().trace_id.value().unwrap();
+        assert!(!trace_id_value.to_string().is_empty());
```

Comment:
This check should still be redundant with the snapshot, afaik the redaction only replaces (not adds missing) values.